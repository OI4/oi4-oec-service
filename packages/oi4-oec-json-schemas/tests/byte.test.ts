import { matchersWithOptions } from 'jest-json-schema'
import schema from '../schemas/dataTypes/byte.schema.json'

expect.extend(
  matchersWithOptions({
    verbose: true,
  })
)

describe('uint8 schema', () => {
  it('validate schema', () => {
    expect(schema).toBeValidSchema()
  })

  it('test valid values', () => {
    expect(0).toMatchSchema(schema)
    expect(1).toMatchSchema(schema)
    expect(1.0).toMatchSchema(schema)
    expect(254).toMatchSchema(schema)
    expect(255).toMatchSchema(schema)
  })

  it('test invalid values', () => {
    expect(-1).not.toMatchSchema(schema)
    expect(1.5).not.toMatchSchema(schema)
    expect(256).not.toMatchSchema(schema)

    expect(() => {
      expect(-1).toMatchSchema(schema)
    }).toThrowErrorMatchingSnapshot('-1')
    expect(() => {
      expect(1.5).toMatchSchema(schema)
    }).toThrowErrorMatchingSnapshot('1.5')
    expect(() => {
      expect(256).toMatchSchema(schema)
    }).toThrowErrorMatchingSnapshot('256')
  })
})
