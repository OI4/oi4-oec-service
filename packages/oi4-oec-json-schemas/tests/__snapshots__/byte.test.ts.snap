// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uint8 schema test invalid values: -1 1`] = `
"expect(received).toMatchSchema(schema)

received
  must be >= 0
    Path:     dataTypes/byte.schema.json#/minimum
"
`;

exports[`uint8 schema test invalid values: 1.5 1`] = `
"expect(received).toMatchSchema(schema)

received
  must be integer
    Path:     dataTypes/byte.schema.json#/type
"
`;

exports[`uint8 schema test invalid values: 256 1`] = `
"expect(received).toMatchSchema(schema)

received
  must be <= 255
    Path:     dataTypes/byte.schema.json#/maximum
"
`;
